resources/framework.rb

```diff
@@ -17,15 +17,121 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-actions :install
-default_action :install
+
 # The provides method is available on chef >= 12.0 only
 provides :ms_dotnet_framework, os: 'windows' if respond_to?(:provides)
 
-attribute :feature_source,  default: nil,         kind_of: [String, nil]
-attribute :include_patches, default: true,        kind_of: [TrueClass, FalseClass]
-attribute :package_sources, default: {}.freeze,   kind_of: Hash
-attribute :perform_reboot,  default: false,       kind_of: [TrueClass, FalseClass]
-attribute :require_support, default: false,       kind_of: [TrueClass, FalseClass]
-attribute :timeout,         default: 600,         kind_of: Integer
-attribute :version,         name_attribute: true, kind_of: String
+property :feature_source,  String
+property :include_patches, [true, false], default: true
+property :package_sources, Hash, default: {}.freeze
+property :perform_reboot, [true, false], default: false
+property :require_support, [true, false], default: false
+property :timeout, Integer, default: 600
+property :version, String, name_property: true
+
+load_current_value do |desired|
+  version_helper = ::MSDotNet.version_helper node, desired.version.to_i
+  version version_helper.installed_version unless version_helper.installed_version.nil?
+end
+
+action :install do
+  if unsupported?
+    ::Chef::Log.info "Unsupported .NET version: #{new_resource.version}"
+    raise "Can't install unsupported .NET version `#{new_resource.version}'" if new_resource.require_support
+  elsif install_required?
+    # Handle features
+    converge_by("Installing .NET version: #{new_resource.version}") do
```

Comment:
Why are you using `converge_by` here? I'm not against it, I just don't understand the point here. `install_features` & `install_packages` are both supporting `why_run`